lms/djangoapps/course_api/api.py

```diff
@@ -38,9 +38,13 @@ def get_effective_user(requesting_user, target_username):
     """
     if target_username == requesting_user.username:
         return requesting_user
+    # This is the default behavior if username is not specified as a query parameter
+    # which is why the is_staff check is happening inside of here.
     elif target_username == '':
+        if requesting_user.is_staff:
+            return requesting_user
         return AnonymousUser()
-    elif can_view_courses_for_username(requesting_user, target_username):
+    elif target_username and can_view_courses_for_username(requesting_user, target_username):
```

Comment:
Do you need this `target_username and` bit if you check for empty string above?  Oh maybe it can be None or whatnot?

Reply:
Yeah, it's basically for the None case (which I don't know if it can technically happen). I ran into it in my tests when I tried passing in None, but then the `can_view_courses_for_username` will throw  an error if target_username is not defined. So I  just make sure we don't hit that error and would raise a PermissionDenied  instead.